include/mitsuba/render/endpoint.h

```diff
@@ -115,6 +115,52 @@ class MTS_EXPORT_RENDER Endpoint : public Object {
                      const Point2f &sample,
                      Mask active = true) const;
 
+    /**
+     * \brief Independently of the scene, sample a position on the endpoint.
+     *
+     * This is entirely independent of transport phenomena, and samples
+     * uniformly over the surface area.
+     *
+     * For infinite area lights, the returned point is placed on a bounding
+     * sphere around the scene. However, since this light is emitted in parallel
+     * rays from infinitely far, it should be interpreted as a direction.
+     * This case can be detected with \ref Emitter::is_environment.
+     *
+     * The default implementation throws an exception.
+     *
+     * \param sample
+     *     A uniformly distributed 2D point on the domain <tt>[0,1]^2</tt>
+     *
+     * \return
+     *     A \ref PositionSample instance describing the generated sample
+     *     along with a spectral importance weight.
+     */
+    virtual std::pair<PositionSample3f, Float>
+    sample_position(Float time, const Point2f &sample,
+                    Mask active = true) const;
+
+    /**
+     * \brief Importance sample a set of wavelengths proportional to the
+     * sensitivity/emission spectrum.
```

Comment:
You need a newline here, otherwise the following text gets merged into the `\brief` part.